sklearn/linear_model/_coordinate_descent.py

```diff
@@ -1649,8 +1658,20 @@ def fit(self, X, y, sample_weight=None):
                 for l1_ratio in l1_ratios
             ]
         else:
+            # Making sure alphas entries are scalars.
+            if isinstance(alphas, (np.ndarray, list, tuple)):
+                n_alphas = 1 if np.ndim(alphas) == 0 else len(alphas)
+                if n_alphas != 1:
+                    for index, alpha in enumerate(alphas):
+                        alpha = check_scalar_alpha(alpha, f"alphas[{index}]")
+                else:
+                    alphas[0] = check_scalar_alpha(alphas[0], "alphas")
+            else:
+                # check for single non-iterable item.
+                alphas = check_scalar_alpha(alphas, "alphas")
```

Comment:
Eventually, we could change the order of the tests here to use [`np.isscalar`](https://numpy.org/doc/stable/reference/generated/numpy.isscalar.html) as mentioned by @thomasjpfan in https://github.com/scikit-learn/scikit-learn/pull/21303#discussion_r813995734.
```suggestion
            if np.isscalar(alphas):
                alphas = check_scalar_alpha(alphas, "alphas")
            else:
                # alphas is an iterable item in this case.
                n_alphas = 1 if np.ndim(alphas) == 0 else len(alphas)
                if n_alphas != 1:
                    for index, alpha in enumerate(alphas):
                        alpha = check_scalar_alpha(alpha, f"alphas[{index}]")
                else:
                    alphas[0] = check_scalar_alpha(alphas[0], "alphas")
```